core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -154,6 +155,33 @@ static HttpResponse delayed(HttpResponse response, Duration delay, ScheduledExec
         return res;
     }
 
+    /***
+     * Invokes the specified {@link Supplier} and creates a new HTTP response that
+     * delegates to the provided {@link HttpResponse} by {@link Supplier},
+     * beginning publishing after {@code delay} has passed from a random {@link ScheduledExecutorService}.
+     */
+    static HttpResponse delayed(Supplier<? extends HttpResponse> responseSupplier, Duration delay) {
+        requireNonNull(responseSupplier, "responseSupplier");
+        requireNonNull(delay, "delay");
+        return delayed(responseSupplier, delay, CommonPools.workerGroup().next());
+    }
+
+    /***
+     * Invokes the specified {@link Supplier} and creates a new HTTP response that
+     * delegates to the provided {@link HttpResponse} {@link Supplier},
+     * beginning publishing after {@code delay} has passed from the provided {@link ScheduledExecutorService}.
+     */
+    static HttpResponse delayed(Supplier<? extends HttpResponse> responseSupplier,
+                                Duration delay,
+                                ScheduledExecutorService executor) {
```

Comment:
How about migrating `delayed(HttpResponse response, Duration delay, ScheduledExecutorService executor)` to use this?
For example:
```java
delay(() -> response, delay, executor);
```

Reply:
Thank you, let me apply it and make some test cases 🙂